pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -1372,7 +1372,11 @@ func (l *LibvirtDomainManager) GetGuestInfo() (v1.VirtualMachineInstanceGuestAge
 // InterfacesStatus returns the interfaces Guest Agent reported
 func (l *LibvirtDomainManager) InterfacesStatus(domainInterfaces []api.Interface) []api.InterfaceStatus {
 	interfaces := l.agentData.GetInterfaceStatus()
-	return agentpoller.MergeAgentStatusesWithDomainData(domainInterfaces, interfaces)
+	if interfaces != nil {
+		return agentpoller.MergeAgentStatusesWithDomainData(domainInterfaces, interfaces)
+	}
```

Comment:
Can this be embedded in `MergeAgentStatusesWithDomainData`?
As seen here, this is bug prone to assume the caller needs to treat this case.

Reply:
Waiting with this to the sumup of what we are doing with the unit tests / e2e
more info here
https://github.com/kubevirt/kubevirt/pull/5560#issuecomment-830764134

Reply:
The function does a conversion, in case its triggered, 
so it can't just return the input self, see please that there are 2 distinct input types
and one output type.

func MergeAgentStatusesWithDomainData(domInterfaces []api.Interface, interfaceStatuses []api.InterfaceStatus) []api.InterfaceStatus